kernel/Cargo.toml

```diff
@@ -61,6 +66,7 @@ cloud = [
   "object_store/azure",
   "object_store/gcp",
   "object_store/http",
+  "hdfs-native-object-store",
```

Comment:
I don't have a strong opinion on this, but I'd generally want this disabled by default. I think the majority of users wouldn't want this so I'd prefer an opt-in, but @nicklan opinion?

Reply:
Are you suggesting that we have an explicit `hdfs` flag? We had a discussion around not wanting to get overloaded with feature flags for each cloud thing and just put it all under "cloud". I think that's still the right thing to do. It doesn't add any deps beyond `object_store`, so we shouldn't see big compile time increases or anything.

Reply:
Ahh yes, I remember now I said something similar then too, okay then no issue forget about this one.